crates/astria-sequencer/src/accounts/types.rs

```diff
@@ -0,0 +1,114 @@
+use borsh::{
+    BorshDeserialize,
+    BorshSerialize,
+};
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+/// Address represents an account address.
+#[derive(Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Eq, Debug)]
+pub struct Address(String);
+
+impl From<&str> for Address {
+    fn from(s: &str) -> Self {
+        Self(s.to_string())
+    }
+}
+
+impl Address {
+    pub fn to_str(&self) -> &str {
+        &self.0
+    }
+}
+
+// Balance represents an account balance.
+#[derive(
+    Clone,
+    Copy,
+    BorshSerialize,
+    BorshDeserialize,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Debug,
+)]
+pub struct Balance(u128);
+
+impl From<u128> for Balance {
+    fn from(n: u128) -> Self {
+        Self(n)
+    }
+}
+
+impl std::ops::Add<u128> for Balance {
+    type Output = Self;
+
+    fn add(self, rhs: u128) -> Self::Output {
+        Self(self.0 + rhs)
+    }
+}
+
+impl std::ops::Add for Balance {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Self(self.0 + rhs.0)
+    }
+}
+
+impl std::ops::Sub for Balance {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        Self(self.0 - rhs.0)
+    }
+}
+
+impl std::ops::Sub<u128> for Balance {
+    type Output = Self;
+
+    fn sub(self, rhs: u128) -> Self::Output {
+        Self(self.0 - rhs)
+    }
+}
+
+// Nonce represents an account nonce.
+#[derive(
+    Clone,
+    Copy,
+    BorshSerialize,
+    BorshDeserialize,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Debug,
+)]
+pub struct Nonce(u32);
```

Comment:
Should we use a u64 here? I believe ethereum uses [uint64 for nonce](https://github.com/astriaorg/go-ethereum/blob/8b9ba616d1c4fbe2acf80c5f6773ea0446ce801f/core/types/transaction.go#L83), as does [cosmos for sequence](https://docs.cosmos.network/main/core/transactions#:~:text=uint64%20sequence%20%3D%205%3B)?

Reply:
yeah, can update, u64 just seemed a bit large lol

Reply:
Better to plan on being successful than not!